### ISSUE TITLE
Improvements to Connectivity Proxy removal logic

### ISSUE THREAD

**Issue by janmedrek** [kyma-incubator/reconciler#676]
**Description**

Due to our template-apply approach, it might be possible that the Connectivity Proxy reconciler lefts some resources after its removal. Implementing a labeling mechanism to enable precise identification of CP components is needed to preserve the state of the cluster in a good shape.

**Reasons**

We do not want any leftovers in case the component is removed.

**Acceptance Criteria**
- [ ] There are no leftovers after Connectivity Proxy is removed by the reconciler

**Comment by janmedrek** [comment]
I have tried installing and removing the Connectivity Proxy on my cluster, the only observation I have is that `connectivity-proxy-registry-config` Secret in `kyma-system` persists.

There were reports of stuff not being deleted in https://github.tools.sap/kyma/backlog/issues/2214 but I was not able to reproduce it. We should still have that case in mind.

**Comment by janmedrek** [comment]
What would need to be done:
- [ ] Install Connectivity Proxy on the Cluster
- [ ] Remove it
- [ ] Check whether any resources tied to the Connectivity Proxy are abandoned on the cluster
  - `kyma-system` and `istio-system` are namespaces where secrets tend to stay
- [ ] Enhance removal logic to cover those resources

[Remove action in Reconciler](https://github.com/kyma-incubator/reconciler/blob/4b5c4d2ee641944b42a1c26b592275ff3f8f196d/pkg/reconciler/instances/connectivityproxy/command.go#L120)

**Comment by janmedrek** [comment]
I can see the `connectivity-proxy-status` ConfigMap is still present in the `kyma-system` namespace after the Proxy was removed, is this intentionally applied by the reconciler?

**Comment by mvshao** [comment]
Yes, the reconciler intentionally applied `connectivity-proxy-status` ConfigMap during cluster creation/reconciliation. 
Here is a result from the `kubectl get configmap -n kyma-system` command (cluster was created about ~50 minutes ago, Proxy ~5 minutes)

`connectivity-proxy 1      4m24s`
`connectivity-proxy-status                       4      47m`

Content of `connectivity-proxy-status` during reconciliation
```
apiVersion: v1
data:
  last-reconciliation: 2022-05-20 09:27:18.750992714 +0000 UTC m=+89751.808678866
  name: connectivity-proxy
  status: running
  version: ""
kind: ConfigMap
metadata:
  creationTimestamp: "2022-05-20T08:34:35Z"
  labels:
    reconciler.kyma-project.io/managed-by: reconciler
    reconciler.kyma-project.io/origin-version: ""
    reconciler.kyma-project.io/reconciliation-summary: "true"
  name: connectivity-proxy-status
  namespace: kyma-system
  resourceVersion: "28108"
  uid: 18db300c-eb5c-42f8-a10f-93a8e1fa5310
```

Content of `connectivity-proxy-status` after reconciliation
```
apiVersion: v1
data:
  last-reconciliation: 2022-05-20 09:37:52.856416046 +0000 UTC m=+91185.135471038
  name: connectivity-proxy
  status: success
  version: ""
kind: ConfigMap
metadata:
  creationTimestamp: "2022-05-20T08:34:35Z"
  labels:
    reconciler.kyma-project.io/managed-by: reconciler
    reconciler.kyma-project.io/origin-version: ""
    reconciler.kyma-project.io/reconciliation-summary: "true"
  name: connectivity-proxy-status
  namespace: kyma-system
  resourceVersion: "28486"
  uid: 18db300c-eb5c-42f8-a10f-93a8e1fa5310
```
As we can see, this ConfigMap was present before installing the Connectivity Proxy on the cluster and it stores information about the current status of the Proxy.

**Comment by janmedrek** [comment]
@mvshao thanks for the explanation, my question is whether it is ok to let it sit in the cluster since the Proxy **was** there?

I would expect the status to be `removed` or the whole Config Map gone if the Proxy was deleted from the cluster. That would be more "natural" and transparent than having the `success` status and no proxy in the cluster. Is this something set by the Reconciler framework, or under the control of Connectivity Proxy reconciler?

**Comment by mvshao** [comment]
Yes, I agree with your point of view about the transparency of the current cluster status. This ConfigMap is under the control of the Reconciler framework.
After some research, I found [PR](https://github.com/kyma-incubator/reconciler/pull/1089 ) and [this commit](https://github.com/kyma-incubator/reconciler/commit/963a7223ade12bb1753c8987abe999f9bd8dfc34) from 13.05.22 (after our merge) which contains a method for creating a ConfigMap with the `-status` suffix. That is related to all of the components. We need to ask @jeremyharisch if they need this file after the Connectivity Proxy was removed from the cluster.
